src/components/dialog/content/MissingModelsWarning.vue

```diff
@@ -41,20 +41,11 @@ import NoResultsPlaceholder from '@/components/common/NoResultsPlaceholder.vue'
 import { useSettingStore } from '@/stores/settingStore'
 import { isElectron } from '@/utils/envUtil'
 
-// TODO: Read this from server internal API rather than hardcoding here
-// as some installations may wish to use custom sources
-const allowedSources = [
-  'https://civitai.com/',
-  'https://huggingface.co/',
-  'http://localhost:' // Included for testing usage only
-]
-const allowedSuffixes = ['.safetensors', '.sft']
+// Custom models sources can be customized in user settings.
+const allowedSources = await comfyPage.getSetting('Comfy.ModelLibrary.AllowedSources')
```

Comment:
Maybe this should be done on application start to avoid slowing down the dialog display ?

Reply:
accessing settings directly in the code, I think no further preloading is needed.